### PR TITLE
Fix the error when the "train_batch_size" is absent in DeepSpeed config

### PR DESCRIPTION
**Pull request by LZHgrla**
# What does this PR do?

When my deepspeed config does not have `train_batch_size `, this error will occur
```
KeyError: 'train_batch_size'
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/accelerate/utils/dataclasses.py [hunk 1]
```diff
@@ -742,7 +742,7 @@ def set_deepspeed_weakref(self):
                 or ds_config["train_micro_batch_size_per_gpu"] == "auto"
             ):
                 ds_config["train_micro_batch_size_per_gpu"] = 1
-            if ds_config["train_batch_size"] == "auto":
+            if "train_batch_size" in ds_config and ds_config["train_batch_size"] == "auto":
                 del ds_config["train_batch_size"]
```
Comment (muellerzr) [inline src/accelerate/utils/dataclasses.py:746]:
```suggestion
            if ds_config.get("train_batch_size", None) == "auto":
                del ds_config["train_batch_size"]
```

Let's make it a bit cleaner :)

**Review by muellerzr — COMMENTED** [review]
Thanks for your PR! One recommendation to make it a bit cleaner, otherwise LG2M, cc @pacman100

**Review by muellerzr — APPROVED** [review]
Thanks! 🤗

**Review by pacman100 — APPROVED** [review]
Hello @LZHgrla, thank you for the fix, LGTM! 🤗

**Pull request merged**